pallets/tinlake-investor-pool/src/lib.rs

```diff
@@ -740,6 +863,106 @@ pub mod pallet {
 			};
 		}
 
+		pub(crate) fn calculate_collect(
+			loc: TrancheLocator<T::PoolId, T::TrancheId>,
+			order: UserOrder<T::Balance, T::EpochId>,
+			pool: PoolDetails<T::AccountId, T::CurrencyId, T::EpochId, T::Balance>,
+			end_epoch: T::EpochId,
+		) -> OutstandingCollections<T::Balance> {
+			let mut epoch_idx = order.epoch;
+
+			// No collect possible in this epoch
+			if epoch_idx == pool.current_epoch {
```

Comment:
Could we have a `<=` check here? Or why only early return when we have `==`?

Reply:
Returning early here when `==` is because if you submitted a new order in the current epoch, then you can't collect yet; the epoch needs to be executed first. `<=` shouldn't be possible, as you can't submit a new order without collecting first (which reminds me, we need to implement a check for this).